frontend/javascripts/oxalis/model/sagas/isosurface_saga.js

```diff
@@ -296,14 +321,50 @@ function* removeIsosurface(action: RemoveIsosurfaceAction): Saga<void> {
   }
 }
 
+function* noteEveryEditedCell(): Saga<void> {
+  const activeCellId = yield* select(state => enforceVolumeTracing(state.tracing).activeCellId);
+  modifiedCells.add(activeCellId);
+}
+
+function* refreshIsosurfaces(): Saga<void> {
+  const renderIsosurfaces = yield* select(state => state.datasetConfiguration.renderIsosurfaces);
+  if (!renderIsosurfaces) {
+    return;
+  }
+  yield* put(saveNowAction());
+  // We reload all cells that got modified till the start of reloading.
+  // By that we avoid that removing cells that got annotated during reloading from the modifiedCells set.
+  const currentlyModifiedCells = new Set(modifiedCells);
+  modifiedCells.clear();
+  // First create an array containing information about all loaded isosurfaces as the map is manipulated within the loop.
+  for (const [cellId, threeDMap] of Array.from(isosurfacesMap.entries())) {
+    if (!currentlyModifiedCells.has(cellId)) {
+      continue;
+    }
+    const isoSurfacePositions = threeDMap.entries().filter(([value, _position]) => value);
+    if (isoSurfacePositions.length <= 0) {
+      continue;
+    }
+    // Removing Isosurface from cache.
+    yield* call(removeIsosurface, removeIsosurfaceAction(cellId), false);
+    // The isosurface should only be removed once after re-fetching the isosurface first position.
+    let shouldBeRemoved = true;
+    for (const [, position] of isoSurfacePositions) {
+      // Reload the Isosurface.
+      yield* call(ensureSuitableIsosurface, null, position, cellId, shouldBeRemoved);
```

Comment:
I'd add a comment which explains that no redundant work will be done, since the isosurface-loading-code always checks whether a request has already been performed. I myself just stumbled here a bit, since the code requests a new isosurface for each position in the three-d-map, even though `ensureSuitableIsosurface` will also traverse the 3D space. However, your approach is still very correct (since multiple seeds might still be necessary to reconstruct everything). I'd just add the comment so that the next reader does not stumble here :)